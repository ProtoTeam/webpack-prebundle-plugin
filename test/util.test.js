const util = require("../lib/util");

test("should generate entry code", () => {
  const entries = ["react", "react-dom", "@alipay/foo"];
  const external = util.getExternalConfig(entries);
  expect(util.generateEntryJS(entries, external)).toMatchSnapshot();
});
