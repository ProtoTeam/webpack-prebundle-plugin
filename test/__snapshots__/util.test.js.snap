// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate entry code 1`] = `
"const wrapper = (m) => { if (m.default) { m.__esModule = true} return m }
import * as __WEBPACK_PREBUNDLE_GLOBAL_REACT from 'react'
import * as __WEBPACK_PREBUNDLE_GLOBAL_REACTDOM from 'react-dom'
import * as __WEBPACK_PREBUNDLE_GLOBAL_ALIPAYFOO from '@alipay/foo'
window.__WEBPACK_PREBUNDLE_GLOBAL_REACT=wrapper(__WEBPACK_PREBUNDLE_GLOBAL_REACT)
window.__WEBPACK_PREBUNDLE_GLOBAL_REACTDOM=wrapper(__WEBPACK_PREBUNDLE_GLOBAL_REACTDOM)
window.__WEBPACK_PREBUNDLE_GLOBAL_ALIPAYFOO=wrapper(__WEBPACK_PREBUNDLE_GLOBAL_ALIPAYFOO)
"
`;
